src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/custom.py

```diff
@@ -292,12 +303,15 @@ def create_task(client,
         try:
             task = TaskAddParameter.from_dict(json_obj)
         except DeserializationError:
-            tasks = []
             try:
-                for json_task in json_obj:
-                    tasks.append(TaskAddParameter.from_dict(json_task))
-            except (DeserializationError, TypeError):
-                raise ValueError("JSON file '{}' is not formatted correctly.".format(json_file))
+                task_collection = TaskAddCollectionParameter.from_dict(json_obj)
+                tasks = task_collection.value
+            except DeserializationError:
+                try:
+                    for json_task in json_obj:
+                        tasks.append(TaskAddParameter.from_dict(json_task))
+                except (DeserializationError, TypeError):
+                    raise ValueError("JSON file '{}' is not formatted correctly.".format(json_file))
```

Comment:
Overly embedded `try-except`. Is there a way to simplify this piece of logic?

Reply:
Don't necesarily know python sdk/cli standards, but could
1) check type prior to from_dict
2) check first char '[' vs '{'
Probably a couple other implementations as well.

Reply:
Can't think of a more 'elegant' way of doing this . Most other solutions may rid the embedded try's, but involve swallowing the exception just to insert code to check if there was an exception.